Build failure: super-slicer-latest

### Steps To Reproduce

Steps to reproduce the behavior:
1. run `nix build github:nixos/nixpkgs/cd286b21e4acb0d10ba8e8d190abeaebbc59ab1d#super-slicer-latest -L`

### Build log

```
....
super-slicer> In file included from /build/source/src/jpeg-compressor/jpge.cpp:937:
super-slicer> In file included from /nix/store/n15bxkd7id2gvlwpihm4kp675aqkrwmj-glibc-2.40-36-dev/include/stdio.h:970:
super-slicer> /nix/store/n15bxkd7id2gvlwpihm4kp675aqkrwmj-glibc-2.40-36-dev/include/bits/stdio2.h:128:13: error: use of undeclared identifier '__builtin___vfprintf_chk'; did you mean '__builtin___sprintf_chk'?
super-slicer>   128 |   int __r = __builtin___vfprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1,
super-slicer>       |             ^
super-slicer> /nix/store/n15bxkd7id2gvlwpihm4kp675aqkrwmj-glibc-2.40-36-dev/include/bits/stdio2.h:128:13: note: '__builtin___sprintf_chk' declared here
super-slicer> /nix/store/n15bxkd7id2gvlwpihm4kp675aqkrwmj-glibc-2.40-36-dev/include/bits/stdio2.h:128:39: error: cannot initialize a parameter of type 'char *' with an lvalue of type 'FILE *const __restrict' (aka 'jpge::_IO_FILE *const __restrict')
super-slicer>   128 |   int __r = __builtin___vfprintf_chk (__stream, __USE_FORTIFY_LEVEL - 1,
super-slicer>       |                                       ^~~~~~~~
super-slicer> [  4%] Building CXX object src/imgui/CMakeFiles/imgui.dir/imgui.cpp.o
....
```

### Additional context

This can be fixed with a simple patch. I'm very surprised upstream has not done so already

```patch
diff --git a/src/jpeg-compressor/jpge.cpp b/src/jpeg-compressor/jpge.cpp
index f2605ad..8641ed5 100644
--- a/src/jpeg-compressor/jpge.cpp
+++ b/src/jpeg-compressor/jpge.cpp
@@ -30,8 +30,9 @@
 
 #include "jpge.h"
 
-#include <stdlib.h>
-#include <string.h>
+#include <cstdlib>
+#include <cstring>
+#include <cstdio>
 //#include <malloc.h> // not needed, even create bugs in macos
 
 #define JPGE_MAX(a,b) (((a)>(b))?(a):(b))
@@ -934,7 +935,6 @@ namespace jpge {
 	}
 
 	// Higher level wrappers/examples (optional).
-#include <stdio.h>
 
 	class cfile_stream : public output_stream
 	{
```

### Notify maintainers

@cab404 @Moredread @hesiod

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

### Metadata

Please run `nix-shell -p nix-info --run "nix-info -m"` and paste the result.

```console
[user@system:~]$ nix-shell -p nix-info --run "nix-info -m"
 - system: `"x86_64-linux"`
 - host os: `Linux 6.11.3-cachyos, NixOS, 24.11 (Vicuna), 24.11.20241014.a3c0b3b`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.24.9`
 - nixpkgs: `/nix/store/xnjw9gmfmpppdj6bxpw6cfkspc3h6xwl-source`
```

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/applications/misc/prusa-slicer/super-slicer.nix
+++ b/pkgs/applications/misc/prusa-slicer/super-slicer.nix
@@ -17,6 +17,11 @@ let
       hash = "sha256-v0q2MhySayij7+qBTE5q01IOq/DyUcWnjpbzB/AV34c=";
     })
     ./meshboolean-const.patch
+    # Drop if this fix gets merged upstream
+    (fetchpatch {
+      url = "https://github.com/supermerill/SuperSlicer/commit/fa7c545efa5d1880cf24af32083094fc872d3692.patch";
+      hash = "sha256-fh31qrqjQiRQL03pQl4KJAEtbKMwG8/nJroqIDOIePw=";
+    })
   ];
 
   versions = {

